markup/asciidocext/asciidocext_config/config.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 The Hugo Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package asciidoc_config holds asciidoc related configuration.
+package asciidocext_config
+
+// DefaultConfig holds the default asciidoc configuration.
+// These values are asciidoctor cli defaults (see https://asciidoctor.org/docs/user-manual/)
+var (
+	Default = Config{
+		Backend:              "html5",
+		DocType:              "article",
+		Extensions:           []string{},
+		Attributes:           map[string]string{},
+		NoHeaderOrFooter:     true,
+		SafeMode:             "unsafe",
+		SectionNumbers:       false,
+		Verbose:              true,
+		Trace:                false,
+		FailureLevel:         "fatal",
+		WorkingFolderCurrent: false,
+	}
+
+	ExtensionsWhitelist = map[string]bool{
```

Comment:
First, apologies for the tame taken for me to get back to this issue. I had so many "in-progress" notifications that I didn't catch that it was done and ready for review.

Reply:
The Go repository have been doing some renames of their whitelists/blacklists in the light of the recent events in the US, so we might at least try to avoid adding new ones. I suggest we rename these to AllowedExtension and Disallowed*

Reply:
I would prefer if we could unexport these whitelists (e.g. `allowedExtensions`); this would mean that you move all/most of the `parseArg` function to the config package, which would keep all the config handling in one place.

Other than that this PR looks ready to go.

Reply:
@bep what would moving it to the config package look like, in order to avoid cyclic imports?

Reply:
I'm not sure which cyclic imports you're talking about, but given this extract:

```go
var cfg = a.cfg.MarkupConfig.AsciidocExt
	args := []string{}

	if asciidocext_config.BackendWhitelist[cfg.Backend] && cfg.Backend != asciidocext_config.Default.Backend {
		args = append(args, "-b", cfg.Backend)
	}

	for _, extension := range cfg.Extensions {
		if !asciidocext_config.ExtensionsWhitelist[extension] {
			a.cfg.Logger.ERROR.Println("Unsupported asciidoctor extension was passed in. Extension `" + extension + "` ignored.")
			continue
		}

		args = append(args, "-r", extension)
	}
```

The above is basically decoding of configuration and I don't see any packages outside of `asciidocext_config` in the above ...?

Reply:
Asciidocext is an external extension. It would be also nice if we keep this extension self contained and leave parseArgs as private function where it is. What do you think?

Reply:
The parseArg can live where it is, my main point was to keep the "config code" in the config package, if possible.

But we can look at that later, it doesn't matter for the end user.

If you can rebase this against master (I see there is a conflict) and possibly also squash the commits, I will pull this in.

Reply:
The rebase was cumbersome. If the result is fine, please continue.